src/groups/bmq/bmqt/bmqt_uri.cpp

```diff
@@ -186,214 +449,121 @@ Uri::print(bsl::ostream& stream, int level, int spacesPerLevel) const
+void UriParser::initialize(BSLA_UNUSED bslma::Allocator* allocator)
 {
-    bslmt::QLockGuard qlockGuard(&s_initLock);
-
-    // NOTE: We pre-increment here instead of post-incrementing inside the
-    //       conditional check below because the post-increment of an int does
-    //       not work correctly with versions of IBM xlc12 released following
-    //       the 'Dec 2015 PTF'.
-    BSLS_ASSERT(s_initialized <
-                bsl::numeric_limits<bsls::Types::Int64>::max());
-    ++s_initialized;
-    if (s_initialized > 1) {
-        return;  // RETURN
-    }
-
-    const char k_PATTERN[] = "^bmq:\\/\\/"
-                             "(?P<authority>"
-                             "(?P<domain>[-a-zA-Z0-9\\._]*)"
-                             "(?P<tier>\\.~[-a-zA-Z0-9]*)?"
-                             ")/"
-                             "(?P<path>[-a-zA-Z0-9_\\.]*)"
-                             "(?P<q1>\\?(id=)[-a-zA-Z0-9_\\.]+)?$";
-    // NOTE: '~' in the authority is a reserved character, used by domain
-    //       resolver to insert the optional resolved tier (if the last segment
-    //       of a 'domain' starts by '~', this means it represents the tier and
-    //       was automatically added by the domain resolver).
-    // NOTE: we use '*' and not '+' for the authority and path parts so that
-    //       when parsing the URI, we can return more detailed error
-    //       (i.e. empty 'authority', instead of a generic regular expression
-    //       mismatch).
-
-    bsl::string error;
-    size_t      errorOffset;
-
-    new (s_regex.buffer())
-        bdlpcre::RegEx(bslma::Default::globalAllocator(allocator));
-    // Enable JIT compilation, unless running under MemorySanitizer.
-    // Low-level assembler instructions used by sljit causes sanitizer issues.
-    // See the internal ticket 177953779.
-    int regexOptions = bdlpcre::RegEx::k_FLAG_JIT;
-#if defined(__has_feature)  // Clang-supported method for checking sanitizers.
-#if __has_feature(memory_sanitizer)
-    regexOptions &= ~bdlpcre::RegEx::k_FLAG_JIT;
-#endif
-#elif defined(__SANITIZE_MEMORY__)  // GCC-supported macros for checking MSAN.
-    regexOptions &= ~bdlpcre::RegEx::k_FLAG_JIT;
-#endif
-
-    int rc = s_regex.object().prepare(&error,
-                                      &errorOffset,
-                                      k_PATTERN,
-                                      regexOptions);
-    if (rc != 0) {
-        BALL_LOG_ERROR << "#URI_REGEXP "
-                       << "Failed to compile URI regular expression [error: '"
-                       << error << "', offset: " << errorOffset << "]";
-    }
-    BSLS_ASSERT_OPT(rc == 0 && "Failed to compile URI regular expression");
+    // NOTHING
 }
 
 void UriParser::shutdown()
 {
-    bslmt::QLockGuard qlockGuard(&s_initLock);
-
-    BSLS_ASSERT(s_initialized > 0);
-    if (--s_initialized != 0) {
-        return;  // RETURN
-    }
-
-    s_regex.object().~RegEx();
```

Comment:
Also note that `UriParser::parse` is not protected by mutex and it checks and uses `s_regex.object()` in a thread unsafe manner. Possible to call `UriParser::shutdown` and `UriParser::parse` from 2 different threads and cause a crash